build-tools/build.sh

```diff
@@ -0,0 +1,248 @@
+#!/bin/bash
+script_start_time=$SECONDS
+# ------------
+# ECHO LICENSE
+# ------------
+echo "::group::License"
+cat /LICENSE
+echo "::endgroup::"
+
+# ----------------
+# SETTING VARIABLES
+# ----------------
+
+set -e
+
+git config --global --add safe.directory /github/workspace
+
+# Create a github group
+echo "::group::Debug Variables"
+
+# -------------------
+# DEBUGGING VARIABLES
+# -------------------
+
+# Echo all variables starting with GITHUB_ for debugging
+for var in "${!GITHUB_@}"; do
+    echo "$var=${!var}"
+done
+
+# Echo all variables starting with INPUT_ for debugging
+for var in "${!INPUT_@}"; do
+    echo "$var=${!var}"
+done
+
+# echo all variables starting with RUNNER_ for debugging
+for var in "${!RUNNER_@}"; do
+    echo "$var=${!var}"
+done
+
+
+echo "::endgroup::"
+# ----------------
+# CHECK IF TEMPLATE
+# ----------------
+echo "::group::Checking if this is a template"
+
+template=$(awk -F'=' '/^IS_LIBRARY:=/{print $2}' Makefile)
+if [ "$template" == "1" ]; then
+    echo "is template"
+else
+    echo "is not template"
+fi
+echo "template=$template" >> $GITHUB_OUTPUT
+
+echo "::endgroup::"
+# ----------------
+# GET PROJECT INFO
+# ----------------
+echo "::group::Getting project info"
+
+if [ "$GITHUB_EVENT_NAME" == "pull_request" ]; then
+    # Fetch the head SHA directly from the PR API
+    API_URL="https://api.github.com/repos/$GITHUB_REPOSITORY/pulls/$GITHUB_PR_NUM"
+    echo "API URL: $API_URL"
+    API_RESPONSE=$(wget -O- --quiet "$API_URL")
+    if [ $? -ne 0 ]; then
+        echo "Error fetching data from GitHub API"
+        exit $?
+    fi
+    echo "API Response: $API_RESPONSE"
+    sha=$(echo "$API_RESPONSE" | jq -r '.head.sha' | head -c 6)
+
+else
+    # Use the commit SHA after the event
+    sha=$(git rev-parse HEAD | head -c 6)
+fi
+
+
+
+function get_sha() {
+    echo "sha=$sha" >> $GITHUB_OUTPUT
+    echo "SHA found: $sha"
+}
+
+function get_version() {
+    version=$(awk -F'=' '/^VERSION:=/{print $2}' Makefile)
+    echo "Version found: $version"
+    echo "version=$version" >> "$GITHUB_OUTPUT"
+    echo $version >> "version.txt"
+}
+
+function get_library_name() { 
+    library_name=$(awk -F'=' '/^LIBNAME:=/{print $2}' Makefile)
+    echo "library_name=$library_name" >> "$GITHUB_OUTPUT"
+    echo "Library name found: $library_name"
+    echo $library_name >> "library_name.txt"
+}
+
+get_sha &
```

Comment:
How much time does this actually save?

Reply:
Probably not much

Reply:
This probably spends more time than it would normally, as the os has to manage the scheduling of the processes. Also why put them in files? Just put them in variables, no?